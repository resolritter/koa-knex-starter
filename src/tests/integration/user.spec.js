const { initLibUser, snapshot } = require("./lib/user")
const { serviceOrchestrationWrapper } = require("./lib/utils")

const tests = [
  [
    "Creates a user and updates it",
    async function () {
      const { createUser, updateUser, loginWithUser } = initLibUser(
        ...arguments,
      )

      let user = await createUser()
      snapshot(user)

      user = await loginWithUser(user)
      snapshot(user)

      user = await updateUser({
        ...user,
        email: `updated_${user.email}`,
      })
      snapshot(user)
    },
  ],
]

describe("Tests the user API", function () {
  tests.map(function ([title, test]) {
    it(title, serviceOrchestrationWrapper(test))
  })
})
