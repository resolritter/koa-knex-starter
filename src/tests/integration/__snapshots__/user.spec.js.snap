// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the user API Creates a user and updates it 1`] = `
Object {
  "email": "user1@user.com",
}
`;

exports[`Tests the user API Creates a user and updates it 2`] = `
Object {
  "access_mask": null,
  "email": "user1@user.com",
  "id": 1,
}
`;

exports[`Tests the user API Creates a user and updates it 3`] = `
Object {
  "access_mask": null,
  "email": "updated_user1@user.com",
  "id": 1,
}
`;
